src/types.d.ts

```diff
@@ -0,0 +1,11 @@
+import { Component, ComponentType } from 'react';
+
+// Removes key type K from object type T
+export type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;
```

Comment:
Could you explain what's going on in this file?

Reply:
These are utility types that I needed to type the spreaded props. Omit is used to exclude the ref prop (it's incompatible when passing between components because they have different instance types) and ExtractProps is used to get the props of an existing component (so I don't have to use the verbose StyledComponentProps type).